pkg/trace/api/api.go

```diff
@@ -451,17 +451,24 @@ func (r *HTTPReceiver) handleStats(w http.ResponseWriter, req *http.Request) {
 func (r *HTTPReceiver) handleTraces(v Version, w http.ResponseWriter, req *http.Request) {
 	ts := r.tagStats(v, req.Header)
 	tracen, err := traceCount(req)
-	if err == nil && r.rateLimited(tracen) {
+	if err == errInvalidHeaderTraceCountValue {
+		log.Errorf("Failed to count traces: %s", err)
+	}
+	defer req.Body.Close()
+
+	select {
+	case r.recvsem <- struct{}{}:
+	case <-time.After(time.Duration(r.conf.DecoderTimeout) * time.Millisecond):
 		// this payload can not be accepted
 		io.Copy(io.Discard, req.Body) //nolint:errcheck
-		w.WriteHeader(r.rateLimiterResponse)
+		w.WriteHeader(http.StatusTooManyRequests)
```

Comment:
Do we want to change this? I remember us chatting briefly about how some old tracers might rely on the agent to _not_ send a 429 response (Although I think if we _can_ move to 429 instead of some weird config option we should definitely do so)

Reply:
My thinking is that we should push for this. We *may* have to go back and use the config, but I think enough time has passed from the initial implementation that we should at least revisit and heavily push that this is something tracers need to handle.

Definitely needs discussion and buy-in from the tracer teams though.

Reply:
This has been moved into an opt-in header.